plugins/formal-verification/src/org/jetbrains/kotlin/formver/conversion/ContractDescriptionConversionVisitor.kt

```diff
@@ -0,0 +1,74 @@
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.formver.scala.silicon.ast.Exp
+import org.jetbrains.kotlin.formver.scala.silicon.ast.Type
+
+class ContractDescriptionConversionVisitor : KtContractDescriptionVisitor<Exp, StmtConversionContext, ConeKotlinType, ConeDiagnostic>() {
+    override fun visitConstantDescriptor(
+        constantReference: KtConstantReference<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        val retType = data.methodCtx.returnVar.type.viperType
+        return when (constantReference) {
+            ConeContractConstantValues.WILDCARD -> Exp.BoolLit(true)
+            ConeContractConstantValues.NULL -> Exp.EqCmp(Exp.LocalVar("ret\$", retType), Exp.NullLit())
+            ConeContractConstantValues.NOT_NULL -> Exp.NeCmp(Exp.LocalVar("ret\$", retType), Exp.NullLit())
+            ConeContractConstantValues.TRUE -> Exp.EqCmp(Exp.LocalVar("ret\$", retType), Exp.BoolLit(true))
+            ConeContractConstantValues.FALSE -> Exp.EqCmp(Exp.LocalVar("ret\$", retType), Exp.BoolLit(false))
+            else -> throw Exception("Unexpected constant: $constantReference")
+        }
+    }
+
+    override fun visitReturnsEffectDeclaration(
+        returnsEffect: KtReturnsEffectDeclaration<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        return returnsEffect.value.accept(this, data)
+    }
+
+    override fun visitBooleanValueParameterReference(
+        booleanValueParameterReference: KtBooleanValueParameterReference<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        // TODO: find a better way to do that
+        return Exp.LocalVar("local\$${booleanValueParameterReference.name}", Type.Bool)
+    }
+
+    override fun visitLogicalBinaryOperationContractExpression(
+        binaryLogicExpression: KtBinaryLogicExpression<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        val left = binaryLogicExpression.left.accept(this, data)
+        val right = binaryLogicExpression.right.accept(this, data)
+        return when (binaryLogicExpression.kind) {
+            LogicOperationKind.AND -> Exp.And(left, right)
+            LogicOperationKind.OR -> Exp.Or(left, right)
+        }
+    }
+
+    override fun visitConditionalEffectDeclaration(
+        conditionalEffect: KtConditionalEffectDeclaration<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        val effect = conditionalEffect.effect.accept(this, data)
+        val cond = conditionalEffect.condition.accept(this, data)
+        return Exp.Implies(effect, cond)
+    }
+
+    override fun visitEffectDeclaration(
+        effectDeclaration: KtEffectDeclaration<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        return effectDeclaration.accept(this, data)
+    }
+
+}
```

Comment:
I think `KtBooleanConstantReference` and `KtLogicalNot` are probably also immediately worth implementing.

Reply:
yes, I was planning to add `KtLogicalNot` after this PR (because I also need to add the negation expr in `org.jetbrains.kotlin.formver.scala.silicon.ast.exp`)

Reply:
i think that `KtBooleanConstantReference` are already captured by `visitConstantDescriptor`

Reply:
forget it, I understood the mistake